python/src/phylanx.cpp

```diff
@@ -501,23 +547,82 @@ PYBIND11_MODULE(_phylanx, m)
-                            phylanx::ir::node_data<double>{d})};
-                });
+                    phylanx::ir::node_data<double>{d})};
+            });
         },
         "create a new variable from a floating point value");
+    execution_tree.def("var",
+        [](const std::vector<double>& d) {
+            return hpx::threads::run_as_hpx_thread([&]() {
+                using namespace phylanx::execution_tree;
+                return primitive{hpx::local_new<primitives::variable>(
+                    phylanx::ir::node_data<double>{d})};
+            });
+        },
+        "create a new variable from a vector floating point values");
+    execution_tree.def("var",
+        [](const std::vector<std::vector<double>>& d) {
+            return hpx::threads::run_as_hpx_thread([&]() {
+                using namespace phylanx::execution_tree;
+                return primitive{hpx::local_new<primitives::variable>(
+                    phylanx::ir::node_data<double>{d})};
+            });
+        },
+        "create a new variable from a matrix floating point values");
+
+    execution_tree.def(
+        "eval", expression_compiler<>, expression_compiler_help);
+    execution_tree.def("eval",
+        expression_compiler<phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("eval",
+        expression_compiler<
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("eval",
+        expression_compiler<
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("eval",
+        expression_compiler<
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("eval",
+        expression_compiler<
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
+    execution_tree.def("eval",
+        expression_compiler<
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&,
+            phylanx::execution_tree::primitive const&>,
+        expression_compiler_help);
 
```

Comment:
I just realized, that we have now two `eval()` functions. One exposed above, and one exposed by the `primitive` type below. Should this be properly consolidated?